src/server/shell/default_configuration.cpp

```diff
@@ -89,6 +92,27 @@ auto mir::DefaultServerConfiguration::wrap_shell(std::shared_ptr<msh::Shell> con
     return wrapped;
 }
 
+auto mir::DefaultServerConfiguration::the_idle_handler() -> std::shared_ptr<msh::IdleHandler>
+{
+    return idle_handler([this]()
+        {
+            auto const idle_handler = std::make_shared<msh::BasicIdleHandler>(
+                the_idle_hub(),
+                the_input_scene(),
+                the_buffer_allocator(),
+                the_display_configuration_controller());
+
+            auto options = the_options();
+            int const idle_timeout_seconds = options->get<int>(options::idle_timeout_opt);
+            if (idle_timeout_seconds > 0)
+            {
+                idle_handler->set_display_off_timeout(std::chrono::seconds{idle_timeout_seconds});
+            }
```

Comment:
If `idle_timeout_seconds` is negative do we want to quietly ignore it? Or, perhaps, fail noisily?